feat: Add additional codec & channel count configs to Adaptation Set Criteria

dpfister2 opened the pull request:
Close https://github.com/shaka-project/shaka-player/issues/8587

avelad commented:
@dpfister2 Please, see https://github.com/shaka-project/shaka-player/issues/8587#issuecomment-2873031235

avelad commented on lib/player.js:
```diff
@@ -6042,6 +6051,9 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
         codecSwitchingStrategy:
             this.config_.mediaSource.codecSwitchingStrategy,
         audioCodec: codec || '',
+        activeAudioCodec: active.audio.codecs || '',
+        activeAudioChannelCount: active.audio.channelsCount || 0,
```
`active.audio` can be null, you need manage it.

dpfister2 replied:
Updated, looking into preference based criteria now

avelad commented on lib/media/adaptation_set_criteria.js:
```diff
@@ -57,7 +57,10 @@ shaka.media.AdaptationSetCriteria.Factory;
+ *   audioCodec: string,
+ *   activeAudioCodec: string,
+ *   activeAudioChannelCount: number,
+ *   preferredAudioCodecs: !Array<string>
```
Maybe yes

dpfister2 replied:
Added, one thing to note is one config call to adaptation set criteria passed it as the `channelCount`. I updated that to just pass 0 in preference of the new field

dpfister2 commented:
Added both new channel count and audio codec configs to preference based criteria. In both cases we fallback to active codec/channel count if we find no matches for the pre-existing config values. Audio codec also falls back one level further with `preferredAudioCodecs`

avelad commented:
> Added both new channel count and audio codec configs to preference based criteria. In both cases we fallback to active codec/channel count if we find no matches for the pre-existing config values. Audio codec also falls back one level further with `preferredAudioCodecs`

Please add some unit tests

dpfister2 commented:
> Please add some unit tests

Added, and added preferred channel count to pref based criteria. But I had some thought on my implementation for that. The codec fallback makes sense, but applying it to channel count doesn't seem right. If a channel count isn't provided, does it make sense to fall back to active channel count? In the case where the expectation is to not filter anything (by passing 0) this would instead filter by what is variant currently selected.

Does that make sense @avelad? Any thoughts?
